clients/fides-js/rollup.config.mjs

```diff
@@ -28,6 +30,14 @@ export default [
         verbose: true,
         hook: "writeBundle",
       }),
+      copy({
+        // Automatically add the built css to the privacy center's static files for bundling:
+        targets: [
+          { src: `dist/${name}.css`, dest: "../privacy-center/public/lib/" },
+        ],
+        verbose: true,
```

Comment:
I agree that it should work like this, but I've been wrangling with this for a while, and doesn't seem to work in the way we expect. There must be some sort of tsconfig setting I'm missing. I'll track in a follow-up

Reply:
Totally fine. I'd be OK with either:
1. Bundling the CSS as a separate `fides.css` output, and updating our docs so customers import it directly
2. OR, bundling the CSS into the `fides.js` itself with rollup

I don't love this current setup though, where we're inserting the CSS at the last minute in the bundle on the privacy center. That feels very error-prone, so let's create a follow-up ticket and tackle that.

Reply:
to be clear, rollup already bundles the CSS into a separate `fides.css` output in `fides-js/dist/fides.css`. I've added the above lines to copy it over into `privacy-center/public/lib/fides.css` so that `privacy-center/public/fides-js-components-demo.html` can import it directly in the HTML. This file is only used by cypress tests.

For our original `privacy-center/public/fides-js-demo.html` file, we dynamically insert the css from `fides-js/dist/fides.css`,  just like we do the js file from `fides-js/dist/fides.js`

Reply:
Yup!

I did a bit of research and found that several other rollup plugins exist that support the ability to automatically "inject" styles into the `<head>`. npm tends to be a bit of a popularity contest so finding a good module that is popular and well-maintained tends to be the safest bet.

After poking around a bit I found these three:
* `rollup-plugin-postcss` (700k weekly downloads, but unmaintained - last release Nov 2021)
* `rollup-plugin-styles` (53k weekly downloads, also unmaintained - last release Jan 2022)
* `rollup-plugin-import-css` (21k weekly downloads, maintained - last release Mar 2023) <-- this is the one you're using

While *most* of the articles online push towards the postcss plugin, I think we'd be OK to stick with this `import-css` plugin and just use the `document.adoptedStylesheets` API: https://developer.mozilla.org/en-US/docs/Web/API/Document/adoptedStyleSheets

That should be as simple as changing this to be:
```
import styles from "./styles.css" assert { type: "css" };

document.adoptedStyleSheets = [styles];
```

Reply:
I did see those other plugins, but opted for mine as it was the most lightweight AND also fairly well maintained.

I didn't know about `document.adoptedStyleSheets` at all, thanks! I'll try this out

Reply:
Hm, so this isn't working for me. Here's what I did: 

1. Created a `declaration.d.ts` file with `declare module "*.module.css";` in `fides-js/src`
2. Added the below to `ConsentBanner.tsx`:

```
import styles from "../lib/banner.module.css" assert { type: "css" };

if (typeof window !== 'undefined') {
  // @ts-ignore
  // Create an empty "constructed" stylesheet
    const sheet = new CSSStyleSheet();
  // Apply a rule to the sheet
  sheet.replaceSync(JSON.stringify(styles));
  document.adoptedStyleSheets = [sheet];
}
```

3. Removed the extra css imports from `privacy-center/public/fides-js-components-demo.html` and `privacy-center/pages/api/fides-js.ts`

This did bundle css into `dist/fides.js`, but the actual css styles were not reflected on the demo page
<img width="813" alt="Screenshot 2023-05-10 at 12 18 02 PM" src="https://github.com/ethyca/fides/assets/7697292/98f7b9bb-3fe9-4d95-9a74-09900dc4212a">

Reply:
tracking follow-up here- https://github.com/ethyca/fides/issues/3288

Reply:
👍 FWIW @eastandwestwind this'll be a good amount of trial-and-error, as most build system jiggery pokery tends to be, so I do think it's best to do in a singular ticket & PR.